codegen/config/binder.go

```diff
@@ -414,12 +414,15 @@ func (b *Binder) TypeReference(schemaType *ast.Type, bindTarget types.Type) (ret
 	return nil, fmt.Errorf("%s has type compatible with %s", schemaType.Name(), bindTarget.String())
 }
 
-func (b *Binder) CopyModifiersFromAst(t *ast.Type, usePtr bool, base types.Type) types.Type {
+func (b *Binder) CopyModifiersFromAst(t *ast.Type, base types.Type) types.Type {
 	if t.Elem != nil {
-		return types.NewSlice(b.CopyModifiersFromAst(t.Elem, usePtr, base))
+		return types.NewSlice(b.CopyModifiersFromAst(t.Elem, base))
 	}
 
-	if !t.NonNull && usePtr {
+	def := b.schema.Types[t.Name()]
+	usePtr := def.Kind != ast.Interface && def.Kind != ast.Union
```

Comment:
@vektah this potentially changes the logic a little bit here, since `usePtr` is recalculated at each recursion, but I think it's maybe more technically correct?  Nothing unexpected seemed to shift after codegen.

Reply:
:+1:  I think its effectively the same thing.